pkg/descheduler/strategies/nodeutilization/highnodeutilization.go

```diff
@@ -0,0 +1,157 @@
+// Note that CPU/Memory requests are used to calculate nodes' utilization and not the actual resource usage.
+func HighNodeUtilization(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, podEvictor *evictions.PodEvictor) {
+	if err := validateNodeUtilizationParams(strategy.Params); err != nil {
+		klog.ErrorS(err, "Invalid HighNodeUtilization parameters")
+		return
+	}
+	thresholdPriority, err := utils.GetPriorityFromStrategyParams(ctx, client, strategy.Params)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get threshold priority from strategy's params")
+		return
+	}
+
+	thresholds := strategy.Params.NodeResourceUtilizationThresholds.Thresholds
+	targetThresholds := strategy.Params.NodeResourceUtilizationThresholds.TargetThresholds
+	if err := validateHighUtilizationStrategyConfig(thresholds, targetThresholds); err != nil {
+		klog.ErrorS(err, "HighNodeUtilization config is not valid")
+		return
+	}
+	targetThresholds = make(api.ResourceThresholds)
+
+	setDefaultForThresholds(thresholds, targetThresholds)
+	resourceNames := getResourceNames(targetThresholds)
+
+	sourceNodes, highNodes := classifyNodes(
+		getNodeUsage(ctx, client, nodes, thresholds, targetThresholds, resourceNames),
+		func(node *v1.Node, usage NodeUsage) bool {
+			return isNodeWithLowUtilization(usage)
+		},
+		func(node *v1.Node, usage NodeUsage) bool {
+			if nodeutil.IsNodeUnschedulable(node) {
+				klog.V(2).InfoS("Node is unschedulable", "node", klog.KObj(node))
+				return false
+			}
+			return !isNodeWithLowUtilization(usage)
+		})
+
+	// log message in one line
+	keysAndValues := []interface{}{
+		"CPU", targetThresholds[v1.ResourceCPU],
+		"Mem", targetThresholds[v1.ResourceMemory],
+		"Pods", targetThresholds[v1.ResourcePods],
+	}
+	for name := range targetThresholds {
+		if !isBasicResource(name) {
+			keysAndValues = append(keysAndValues, string(name), int64(targetThresholds[name]))
+		}
+	}
+
+	klog.V(1).InfoS("Criteria for a node below target utilization", keysAndValues...)
+	klog.V(1).InfoS("Number of underutilized nodes", "totalNumber", len(sourceNodes))
+
+	if len(sourceNodes) == 0 {
+		klog.V(1).InfoS("No node is underutilized, nothing to do here, you might tune your thresholds further")
+		return
+	}
+	if len(sourceNodes) < strategy.Params.NodeResourceUtilizationThresholds.NumberOfNodes {
+		klog.V(1).InfoS("Number of nodes underutilized is less than NumberOfNodes, nothing to do here", "underutilizedNodes", len(sourceNodes), "numberOfNodes", strategy.Params.NodeResourceUtilizationThresholds.NumberOfNodes)
+		return
+	}
+	if len(sourceNodes) == len(nodes) {
+		klog.V(1).InfoS("All nodes are underutilized, nothing to do here")
+		return
+	}
+	if len(highNodes) == 0 {
+		klog.V(1).InfoS("No node is available to schedule the pods, nothing to do here")
+		return
+	}
+
+	evictable := podEvictor.Evictable(evictions.WithPriorityThreshold(thresholdPriority))
```

Comment:
Nit: missing WithNodeFit though it can be added in another PR:
```
evictable := podEvictor.Evictable(evictions.WithPriorityThreshold(thresholdPriority), evictions.WithNodeFit(nodeFit))
```